pkg/daemon/ceph/client/radosnamespace.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/util/exec"
+)
+
+// CreateRadosNamespace create a rados namespace in a pool.
+// poolName is the name of the ceph block pool, the same as the CephBlockPool CR name.
+func CreateRadosNamespace(context *clusterd.Context, clusterInfo *ClusterInfo, poolName, namespaceName string) error {
+	cName := fmt.Sprintf("%s/%s", clusterInfo.Namespace, clusterInfo.name)
+	logger.Infof("%q creating rados namespace %s/%s", cName, poolName, namespaceName)
+	//  rbd namespace create pool-name/namespace-name
+	args := []string{"namespace", "create", "--pool", poolName, "--namespace", namespaceName}
+	cmd := NewRBDCommand(context, clusterInfo, args)
+	cmd.JsonOutput = false
+	output, err := cmd.Run()
+	if err != nil {
+		code, ok := exec.ExitStatus(err)
+		if ok && code == int(syscall.EEXIST) {
+			logger.Infof("%q rados namespace %s/%s already exists", cName, poolName, namespaceName)
```

Comment:
We need to return nil, right? Otherwise an error will still be returned
```suggestion
			logger.Debugf("%q rados namespace %s/%s already exists", cName, poolName, namespaceName)
			return nil
```

Reply:
Yes correct. updated the code